staging/src/k8s.io/client-go/tools/cache/reflector.go

```diff
@@ -100,22 +111,43 @@ func NewNamedReflector(name string, lw ListerWatcher, expectedType interface{},
 		name:          name,
 		listerWatcher: lw,
 		store:         store,
-		expectedType:  reflect.TypeOf(expectedType),
 		period:        time.Second,
 		resyncPeriod:  resyncPeriod,
 		clock:         &clock.RealClock{},
 	}
+	r.setExpectedType(expectedType)
 	return r
 }
 
+func (r *Reflector) setExpectedType(expectedType interface{}) {
+	if expectedType == nil {
+		r.expectedTypeName = defaultExpectedTypeName
```

Comment:
still assign r.expectedType here?

Reply:
Given that `reflect.TypeOf(nil) == nil`,  what would be the value of doing so?

Reply:
Don't get me wrong, I'm happy to make the change.  I'd just like to understand the rational.

Reply:
Done

Reply:
> Don't get me wrong, I'm happy to make the change. I'd just like to understand the rational.

If setExpectedType was previously called, returning early here could leave r.expectedType set to the previous non-nil value

Reply:
Ok, thank you.